service/controller/resource/monitoring/pvcresizingresource/create.go

```diff
@@ -56,6 +61,39 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 					return microerror.Mask(err)
 				}
 				r.logger.Debugf(ctx, "replaced PVC %v", pvc.GetName())
+				needToUpdatePrometheuses = true
+			}
+
+			// PVC have been updated, we need to update Retention.Size field on all prometheuses
+			if needToUpdatePrometheuses {
+				prometheusList, err := r.listPrometheus(ctx)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				for _, prom := range prometheusList.Items {
+					// Patch Retention.Size with a ratio of the desired value
+					patch := []byte(fmt.Sprintf(
+						`{
+							"spec": { 
+								"retentionSize": "%v" 
+							} 
+						}`,
+						promv1.ByteSize(pvcresizing.GetRetentionSize(desiredVolumeSize.AsApproximateFloat64()))))
```

Comment:
Beware that `AsApproximateFloat64()` may return a float, and you may want to give a round number of bytes.
I've had a similar bug recently where I had to cast it to an int.

Ref: https://github.com/giantswarm/prometheus-meta-operator/pull/1187/files#diff-116e2cfdd5bcbff6c2a0fde571e6ce14071cf8f8ac95bead69f1894a9c314366R213